### PR TITLE
Create `MultiTableMetadata` class

### LINKED ISSUE #883: Create MultiTableMetadata class
### Problem Description

_As a user, it would be useful to have a programmatic way to interact with the metadata for all of the tables in my dataset._

### Expected behavior

- In the metadata module, add a class called `MultiTableMetadata`
- The class needs to store a dictionary that maps all of the table names to their `SingleTableMetadata` class instances and another dictionary that stores the relationships.
- The `__init__` requires no parameters
- There should be a `to_dict` method that returns the underlying dictionaries in the proper json format
```python
>>> python_dict = metadata.to_dict()
{
  "tables": {
    "accounts": { <Table Spec> },
    "branches": { <Table Spec> }
  }, 
  "relationships": [
    { <Relationship Spec>},
    { <Relationship Spec> },
  ],
}
```

- There should be an internal `_load_from_dict(dict)` method that returns a class instance with the values set based on a provided python dictionary that matches the json spec. The instance should call the `SingleTableMetadata._load_from_dict` to create the `SingleTableMetadata` instances.
```python
metadata = SingleTableMetadata._load_from_dict(dict)
```

- The `__repr__` method should be overridden to print the full dictionary representation of the json for the `MultiTableMetadata`.
```python
>>> print(metadata)
{
  "tables": {
    "accounts": { <Table Spec> },
    "branches": { <Table Spec> }
  }, 
  "relationships": [
    { <Relationship Spec>},
    { <Relationship Spec> },
  ],
}
```

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #883

### CODE REVIEW

File: sdv/metadata/single_table.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+
+        This method automatically detects the ``sdtypes`` for the given ``pandas.DataFrame``.
+
+        Args:
+            data (pd.DataFrame):
+                ``pandas.DataFrame`` to detect the metadata from.
+        """
+        if self._columns:
+            raise ValueError(
+                'Metadata already exists. Create a new ``SingleTableMetadata`` '
```
Comment (fealho) [inline sdv/metadata/single_table.py:210]: There are a lot of `SingleTableMetadata` around the code, maybe just crtrl+f for them and substitute with `MultiTableMetadata`.

File: tests/unit/metadata/test_multi_table.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        assert instance._relationships == []
+
+    def test_to_dict(self):
+        """Test the ``to_dict`` method of ``MultiTableMetadata``.
+
+        Setup:
+            - Instance of ``MultiTableMetadata``.
+            - Add mocked values to ``instance._tables`` and ``instance._relationships``.
+        Mock:
+            - Mock ``SingleTableMetadata`` like object to ``instance._tables``.
```
Comment (fealho) [inline tests/unit/metadata/test_multi_table.py:27]: There are a lot of `SingleTableMetadata` in the multitable files. Maybe ctrl f for them and put `MultiTableMetadata` instead.

File: tests/unit/metadata/test_multi_table.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    @patch('sdv.metadata.multi_table.SingleTableMetadata')
+    def test__set_metadata(self, mock_singletablemetadata):
+        """Test the ``_set_metadata`` method for ``MultiTableMetadata``.
+
+        Setup:
+            - instance of ``MultiTableMetadata``.
+            - A dict representing a ``MultiTableMetadata``.
+
+        Mock:
+            - Mock ``SingleTableMetadata`` from ``sdv.metadata.multi_tbale``
```
Comment (fealho) [inline tests/unit/metadata/test_multi_table.py:95]: `table`

File: tests/unit/metadata/test_multi_table.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        """Test that ``_load_from_dict`` returns a instance of ``MultiTableMetadata``.
+
+        Test that when calling the ``_load_from_dict`` method a new instance with the passed
+        python ``dict`` details should be created.
+
+        Setup:
+            - A dict representing a ``MultiTableMetadata``.
+
+        Mock:
+            - Mock ``SingleTableMetadata`` from ``sdv.metadata.multi_tbale``
```
Comment (fealho) [inline tests/unit/metadata/test_multi_table.py:160]: `table`

**Review by amontanez24 — APPROVED** [review]
Besides @fealho 's comments this looks good to me. After addressing :shipit:

File: tests/unit/metadata/test_multi_table.py (hunk 2 above)
Comment (pvk-developer) [reply tests/unit/metadata/test_multi_table.py:27]: I think they should be ok since they are mocks of `SingleTableMetadata` or just calls from it

**Review by fealho — APPROVED** [review]

**Pull request merged**